File: backend/Entities/TaskItem.cs
namespace Jamrah.Backend.Entities;

public class TaskItem
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? GoalId { get; set; }
    public int Completed { get; set; }
    public string CreatedAt { get; set; } = string.Empty;
    public string? ParentTaskId { get; set; }
    public string Priority { get; set; } = "none";
    public string? CompletedAt { get; set; }
    public string? ScheduledTime { get; set; }
    public string? Recurrence { get; set; }
    public string? CustomDays { get; set; }
    public string? DurationStart { get; set; }
    public string? DurationEnd { get; set; }
    public string? Notes { get; set; }

    public Goal? Goal { get; set; }
    public TaskItem? ParentTask { get; set; }
    public ICollection<TaskItem> Subtasks { get; set; } = new List<TaskItem>();
}

File: android/app/src/main/java/com/jamrah/app/domain/model/TaskItem.kt
package com.jamrah.app.domain.model

/**
 * Domain model for a Task.
 * Matches the PC backend entity exactly:
 *   - String IDs matching format "task_{timestamp}_{random}"
 *   - completed as Int (0 = pending, 1 = done) — matches SQLite default
 *   - priority as String ("none" | "Low" | "Medium" | "High")
 *   - recurrence as String ("none" | "daily" | "weekly" | "monthly" | "custom")
 *   - customDays as JSON string e.g. "[0,1,4]" (day-of-week indices, Sun=0)
 *   - scheduledTime as "yyyy-MM-dd" or "yyyy-MM-ddTHH:mm"
 */
data class TaskItem(
    val id: String,
    val name: String,
    val goalId: String? = null,
    val completed: Int = 0,
    val createdAt: String,
    val parentTaskId: String? = null,
    val priority: String = "none",
    val completedAt: String? = null,
    val scheduledTime: String? = null,
    val recurrence: String? = null,
    val customDays: String? = null,
    val durationStart: String? = null,
    val durationEnd: String? = null,
    val notes: String? = null,

    // Transient — set locally when expanding recurring tasks
    val instanceDate: String? = null,
    val baseId: String? = null
)